app/services/ai_backend/anthropic.rb

```diff
@@ -28,6 +28,19 @@ def get_next_chat_message(&chunk_received_handler)
 
     response_handler = proc do |intermediate_response, bytesize|
       chunk = intermediate_response.dig("delta", "text")
+
+      input_tokens = intermediate_response.dig("message", "usage", "input_tokens")
+      if input_tokens
+        @message.input_token_count += input_tokens
+      end
+      # the initial number of output tokens is first returned in message.usage.output_tokens and
+      # the rest in usage.output_tokens in the final response
+      output_tokens = intermediate_response.dig("message", "usage", "output_tokens")
+      output_tokens = intermediate_response.dig("usage", "output_tokens") if (output_tokens.nil?)
```

Comment:
you can lose the `if` at the end of the line and instead to:

```suggestion
      output_tokens ||= intermediate_response.dig("usage", "output_tokens")
```